src/components/Identity/index.js

```diff
@@ -110,12 +99,8 @@ const createIdentity = ({ config, logger, cookieJar }) => {
             deferredForEcid = defer();
             payload.expectResponse();
           }
-          if (haveCustomerIds) {
-            addCustomerIdsToPayload(
-              customerIdState,
-              customerIdChanged,
-              payload
-            );
+          if (customerIds.state.hasIds) {
```

Comment:
This feels like we are violating encapsulation of the customerIds' state.  I'd recommend putting this check inside the `addToPayload` method.